apps/pie-docs/src/patterns/cookie-banner/code/code.md

```diff
@@ -0,0 +1,161 @@
+    <img alt="GitHub Workflow Status" src="https://img.shields.io/npm/v/@justeattakeaway/pie-cookie-banner.svg">
+  </a>
+</p>
+
+`pie-cookie-banner` is a Web Component built using the Lit library.
+
+This component can be easily integrated into various frontend frameworks and customized through a set of properties.
+
+## Cookies
+`pie-cookie-banner` purely handles the UI and user interactions for cookie consent. It does not handle the setting or management of cookies or the storage of user preferences. It is the responsibility of the consuming application to handle these aspects. See [Events](#events) for more information.
+
+
+## Installation
+
+To install `pie-cookie-banner` in your application, run the following on your command line:
+
+```shell
+# npm
+$ npm i @justeattakeaway/pie-cookie-banner
+```
+
+```shell
+# yarn
+$ yarn add @justeattakeaway/pie-cookie-banner
+```
+
+{% notification {
+  type: "neutral",
+  iconName: "link",
+  message: "For more information on using PIE components as part of an application, check out the [Getting Started Guide](https://github.com/justeattakeaway/pie/wiki/Getting-started-with-PIE-Web-Components)."
+} %}
+
+## Playground
+
+ <iframe
+  src="https://webc.pie.design/?path=/story/cookie-banner--default&viewMode=story&shortcuts=true&singleStory=true"
+  width="100%"
+  height="900px"
+  style="border: none; margin-top: 32px;"
+></iframe>
+
+
+## Props
+
+{% componentDetailsTable {
+  tableData: props
+} %}
+
+## Events
+
+The PIE cookie banner does not directly manage cookies within the application or component. Instead, it has several events that application developers (consumers of this component) can listen for, letting applications manage their own cookies. The table below provides a list of events you can listen for:
+
+{% componentDetailsTable {
+tableData: events
+} %}
+
+## Localisation
+
+By default, the component displays its content in English. To display the content in another language, you need to import the locale data for that language and pass it in the `locale` prop. For example, to display the content in Dutch, you need to import the Dutch locale data:
+
+```js
+import locale from '@justeattakeaway/pie-cookie-banner/locales/nl-nl.json';
+
+<!-- JSX -->
+<PieCookieBanner locale={locale}></PieCookieBanner>
+```
+
+It's possible to import all locales at once, if necessary:
+
+```js
+import allLocales from '@justeattakeaway/pie-cookie-banner/locales';
+```
+
+## Examples
+
+For HTML:
+
+```js
+// import as module into a js file e.g. main.js
+import '@justeattakeaway/pie-cookie-banner';
+```
+
+```html
+<!-- pass js file into <script> tag -->
+<pie-cookie-banner></pie-cookie-banner>
+<script type="module" src="/main.js"></script>
+```
+
+For Native JS Applications, Vue, Angular, Svelte etc.:
+
+```js
+// Default – for Native JS Applications, Vue, Angular, Svelte, etc.
+import { PieCookieBanner } from '@justeattakeaway/pie-cookie-banner';
```

Comment:
left a comment internally too but we don't recommend object destructing for Vue usage. I would probably reword this example to not include Vue :)

Reply:
Have discussed that we need to solve this perhaps by migrating the [vue wiki guide](https://github.com/justeattakeaway/pie/wiki/PIE-Web-Components-%E2%80%90-Vue-Integration-Guide) to pie.design and linking this to all components. A[ draft ticket](https://justeattakeaway.atlassian.net/browse/DSW-1993) was created to follow this up.